src/test/common.ts

```diff
@@ -59,3 +63,21 @@ export function isBrowser(browsers: Browser | Browser[], ua = window.navigator.u
     return browsersStr.indexOf(browser) > -1;
   }
 }
+
+export function createKeyEvent(key: Key, options?: {type?: 'keyup' | 'keydown'}) {
```

Comment:
Couldn't we write it as `createKeyEvent(key: Key, options?: {type?: 'keyup' | 'keydown'} = {})` so you could drop subsequent lines:

```
if (!isDefined(options)) {
  options = {};
}
```

Reply:
Or better yet: `reateKeyEvent(key: Key, options?: {type?: 'keyup' | 'keydown'} = {type: 'keyup'})`

so we can drop more:

```
let {type} = options;
if (!isDefined(type)) {
  type = 'keyup';
}
```

Reply:
I'm used to write this kind of code - despite the fact it's indeed more verbose - because it tended to be more robust in the general case (for specific cases, we can always adapt).

First, the potential advantage of using `isDefined` instead of the default parameter assignment is to be able to handle not only `undefined` values but also `null` ones.

Then, why not `createKeyEvent(key: Key, options?: {type?: 'keyup' | 'keydown'} = {type: 'keyup'})`, it's because if an empty `options` object is given, we won't have any value for `options.type` anymore.

In both cases we could say that a user shouldn't pass a `null` value or an empty object for `options`, and this would be a contract. That's why I said the code I wrote is more robust, not that it is mandatory in our case (especially that we are dealing with internal, testing code).

Reply:
>  it's because if an empty options object is given, we won't have any value for options.type anymore.

This can be easily fixed with:

```
createKeyEvent(key: Key, options?: {type: 'keyup' | 'keydown'} = {type: 'keyup'})
```

This is shorter,  more explicit and readable, IMO so I would prefer that we use it instead.

Reply:
> In both cases we could say that a user shouldn't pass a null value or an empty object for options, and this would be a contract. 

Agreed. But this is precisely why we should express it in a type - we can explicitly say that you can either omit a params object  or - if you decide to provide one - have it conform to exact specification.